12-watchtower-API.md

```diff
@@ -4,245 +4,337 @@
+The server:
 
-#### Requirements
-The sending node: 
-
-* MUST send `wt_init` as the first message.
-* MUST set `client_server` to the corresponding value, depending on whether it is a client or a server.
-* MUST set `accepted_ciphers` to the list of ciphers he implements.
-* MUST set `modes` to the list of modes that he is willing to accept.
-* SHOULD set `qos` to the the quality of service he is requesting / offering. 
-
-The requesting node: 
-
-* MUST receive `wt_init` before sending any other message.
-* MUST respond with its own `wt_init` message
-* upon receiving a `client_server` field that matches its own (a server receives a `wt_init` from a server or a client receives one from a client):
-	* MUST fail the connection
-* upon receiving an `accepted_ciphers` that does not contain any of its own accepted ciphers:
-	* MUST fail the connection
-* upon receiving a `modes` that does not contain any of its own accepted modes:
-	* MUST fail the connection
-* upon receiving a `qos` that does not contain any of its own accepted QoS:
-	* MUST fail the connection
+* MUST reject the appointment if:
+	* Authentication is required and `auth_token` is not provided.
+	* Authentication is required and `auth_token` is invalid.
+	* `locator` is not a `16-byte` value.
+	* `start_block` is further than one block behind the current chain tip.
+	* `start_block` is further than one block ahead the current chain tip.
+	* `encrypted_blob` has unreasonable size.
+	* `cipher` is not among the ones he implements.
+
+* SHOULD reject the appointment if`end_block` is too far away in the future.
+
+* MUST: 
+	* truncate the remainder of the package to `qos_len`.
+	* process `qos_data` according to [Quality of Service data](#quality-of-service-data) if `qos_len` is not 0.
+
+* MAY accept the appointment otherwise.
 
 #### Rationale
 
-The client is always the one in charge of establishing the connection. His `wt_init` message informs the server of what type of service he is requesting. The server should agree with the client if the implement the same methods. Otherwise the connection should be failed.
+We define appointment as the way that the WatchTower is hired / requested by a client to do it's watching services.
 
-QoS is an optional field. Including the field in the `wt_init` message signals that the sender is requiring that specific QoS. As for accountability, it aims for giving non-repudiable proof of the agreement to both the client and the server.
+WatchTowers may work in different modes (e.g. altruistic vs non-altruistic). In some of those modes, proof of payment may be required for the WatchTower to provide the service. `auth_token` is used to decide whether the user is entitled to use the service or not in the cases it may be required. Notice that the tokens do not need to be linked to any kind of identity, but confirm that the payment has been performed.
```

Comment:
WatchTowers may offer their services for free ('altruistic') or they may require a payment when accepting the job ('non-altruistic'. We have defined 'auth_token' as an authentication mechanism between the client and server, so the client can prove they are entitled to the service. The tokens are not required to be linked to any kind of identity (e.g. blinded token), but their sole purpose is to confirm the client has already paid for the service.

Reply:
Changed